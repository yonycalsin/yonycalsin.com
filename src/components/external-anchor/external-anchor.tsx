import * as React from 'react'
import { Icon, Link, useColorModeValue as mode } from '@chakra-ui/react'
import { FiArrowUpRight } from 'react-icons/fi'

import type { ExternalAnchorProps } from 'typings/components'

function ExternalAnchor(props: ExternalAnchorProps) {
  const { children, ...restProps } = props

  return (
    <span>
      <Link
        {...restProps}
        alignItems="center"
        d="inline-flex"
        color={mode('primary.500', 'primary.300')}
        isExternal
        target="_blank"
        rel="noreferrer"
        role="link"
      >
        {children}
      </Link>
      <Icon as={FiArrowUpRight} d="inline" color={mode('gray.700', 'white')} />
    </span>
  )
}

export default ExternalAnchor
