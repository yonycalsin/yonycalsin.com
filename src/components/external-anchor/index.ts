export { default as ExternalAnchor } from './external-anchor'
