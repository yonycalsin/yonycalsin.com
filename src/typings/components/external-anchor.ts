import type * as React from 'react'
import type { LinkProps } from '@chakra-ui/react'

interface ExternalAnchorProps extends LinkProps {
  children: React.ReactNode
}

export type { ExternalAnchorProps }
