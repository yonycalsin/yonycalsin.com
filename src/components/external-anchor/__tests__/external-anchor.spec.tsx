import { render } from '@testing-library/react'

import { ExternalAnchor } from '../external-anchor'

describe('Components / ExternalAnchor', () => {
  it('renders an external anchor', () => {
    const view = render(<ExternalAnchor href="https://domain.com">Go to home</ExternalAnchor>)

    expect(view.baseElement).toMatchSnapshot()
  })
})
