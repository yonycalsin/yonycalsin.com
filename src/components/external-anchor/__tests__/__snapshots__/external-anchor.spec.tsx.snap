// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalAnchor renders an external anchor 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: primary.500;
}

.emotion-1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: gray.700;
}

<body>
  <div>
    <span>
      <a
        class="chakra-link emotion-0"
        d="inline-flex"
        href="https://domain.com"
        rel="noreferrer"
        target="_blank"
      >
        Go to home
      </a>
      <svg
        class="chakra-icon emotion-1"
        d="inline"
        fill="none"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="7"
          x2="17"
          y1="17"
          y2="7"
        />
        <polyline
          points="7 7 17 7 17 17"
        />
      </svg>
    </span>
  </div>
</body>
`;
